src/Features/Core/Portable/RemoveUnusedMembers/AbstractRemoveUnusedMembersDiagnosticAnalyzer.cs

```diff
@@ -496,6 +603,40 @@ private bool IsEntryPoint(IMethodSymbol methodSymbol)
                    (methodSymbol.ReturnsVoid ||
                     methodSymbol.ReturnType.OriginalDefinition == _taskType ||
                     methodSymbol.ReturnType.OriginalDefinition == _genericTaskType);
+
+            private bool IsMethodWithSpecialAttribute(IMethodSymbol methodSymbol)
+                => methodSymbol.GetAttributes().Any(a => _attributeSetForMethodsToIgnore.Contains(a.AttributeClass));
+
+            private bool IsShouldSerializeOrResetPropertyMethod(IMethodSymbol methodSymbol)
+            {
+                // ShouldSerializeXXX and ResetXXX are ok if there is a matching
+                // property XXX as they are used by the windows designer property grid
+
+                return methodSymbol.ReturnType.SpecialType == SpecialType.System_Boolean  &&
+                    methodSymbol.Parameters.IsEmpty &&
+                    (IsSpecialMethodWithMatchingProperty("ShouldSerialize") ||
+                     IsSpecialMethodWithMatchingProperty("Reset"));
+
+                // Local functions.
+                bool IsSpecialMethodWithMatchingProperty(string prefix)
+                {
+                    if (methodSymbol.Name.StartsWith(prefix))
+                    {
+                        var suffix = methodSymbol.Name.Substring(prefix.Length);
+                        return suffix.Length > 0 &&
+                            methodSymbol.ContainingType.GetMembers(suffix).Any(m => m is IPropertySymbol);
+                    }
+
+                    return false;
+                }
+            }
+
+            private bool IsISerializableConstructor(IMethodSymbol methodSymbol)
+                => _iSerializableType != null &&
+                   methodSymbol.Parameters.Length == 2 &&
+                   methodSymbol.Parameters[0].Type.Equals(_serializationInfoType) &&
+                   methodSymbol.Parameters[1].Type.Equals(_streamingContextType) &&
+                   methodSymbol.ContainingType.AllInterfaces.Contains(_iSerializableType);
```

Comment:
note: not a huge complaint, but it is interesting that you only check one of your special types for null, but you use three of the special types here.  Seems strange.  I am going to assum that .Equals here is well behaved and will not throw if it gets a null symbol passed in, so i think it's ok.

Reply:
Yes, that is correct.